data-prepper-plugins/cloudwatch-metrics-source/src/main/java/org/opensearch/dataprepper/plugins/source/CloudwatchMetricsWorker.java

```diff
@@ -0,0 +1,198 @@
+                            .build();
+
+                    MetricStat metStat = MetricStat.builder()
+                            .stat(metricDataQueriesConfig.getMetricsConfig().getStat())
+                            .unit(metricDataQueriesConfig.getMetricsConfig().getUnit())
+                            .period(metricDataQueriesConfig.getMetricsConfig().getPeriod())
+                            .metric(met)
+                            .build();
+
+                    MetricDataQuery dataQuery = MetricDataQuery.builder()
+                            .metricStat(metStat)
+                            .id(metricDataQueriesConfig.getMetricsConfig().getId())
+                            .returnData(true)
+                            .build();
+
+                    List<MetricDataQuery> dataQueries = new ArrayList<>();
+                    dataQueries.add(dataQuery);
+
+                    GetMetricDataRequest getMetReq = GetMetricDataRequest.builder()
+                            .maxDatapoints(10)
+                            .scanBy(ScanBy.TIMESTAMP_DESCENDING)
+                            .startTime(Instant.parse(namespacesListConfig.getNamespaceConfig().getStartTime()))
+                            .endTime(Instant.parse(namespacesListConfig.getNamespaceConfig().getEndTime()))
+                            .metricDataQueries(dataQueries)
+                            .build();
+
+                    GetMetricDataResponse response = cloudWatchClient.getMetricData(getMetReq);
+                    List<MetricDataResult> data = response.metricDataResults();
+                    writeToBuffer(data, bufferAccumulator, null);
+
+                } catch (CloudWatchException | DateTimeParseException ex) {
+                    LOG.error("Exception Occurred while scraping the metrics  {0}", ex);
+                }
+            }
+        }
+    }
+
+    /**
+     * Helps to write metrics data to buffer and to send end to end acknowledgements after successful processing
+     * @param metricsData metricsData
+     * @param bufferAccumulator bufferAccumulator
+     * @param acknowledgementSet acknowledgementSet
+     */
+    public void writeToBuffer(final List<MetricDataResult> metricsData,
+                              final BufferAccumulator<Record<Event>> bufferAccumulator,
+                              final AcknowledgementSet acknowledgementSet) {
+        metricsData.forEach(message -> {
+            final Record<Event> eventRecord = new Record<Event>(JacksonEvent.fromMessage(message.toString()));
+            try {
+                bufferAccumulator.add(eventRecord);
+            } catch (Exception ex) {
+                LOG.error("Exception while adding record events {0}", ex);
+            }
+            if(Objects.nonNull(acknowledgementSet)){
+                acknowledgementSet.add(eventRecord.getData());
+            }
+        });
+        try {
+            bufferAccumulator.flush();
```

Comment:
You are flushing the buffer accumulator here. As I noted above, you don't even need to BufferAccumulator.

Create a `List<Record<Event>> metricEvents`.

Then call `buffer.writeAll(metricEvents)`.

Reply:
this has been fixed.